Fixed some issues with regex, specifically with displaying captions

jhuang09 opened the pull request:
- Updates to regex:
    - As long as the caption is contained by square brackets, the contents inside will become the caption in the HTML.
    - Image detection (?) portion of the regex can do both .jpg <i>and</i> .jpeg!
- Updates to replacement string:
    - Only the contents inside the square brackets will become the caption of the image. Had to update some of the pattern matching to do so.

jhuang09 committed:
small fix to make regex to be less buggy

jhuang09 committed:
made small changes to regex and replacement string

jhuang09 committed:
made small changes to regex and replacement string

jhuang09 committed:
update pattern matching

jhuang09 committed:
Merge branch 'master' into JannyDevelopment

jhuang09 committed:
update pattern matching

jhuang09 committed:
added tests to MessageServletTest.java

jhuang09 committed:
Merge branch 'JannyDevelopment' of https://github.com/mrios730/Team-25 into JannyDevelopment

jhuang09 committed:
fixed linter errors

dhgarrette reviewed (approved):
Cool!  If you were up for it, you could probably add a few more test cases just to make sure that things don't change unexpectedly in the future.  Here are a couple ideas:
* Empty string.
* Input without any URLs.
* Input with multiple image URLs.
* Input with text before the caption.
* Input with something in square brackets before the caption, i.e. `[text1] [text2] URL`.

jhuang09 committed:
Added more tests and modified regex to be more foolproof

jhuang09 commented:
I added more tests! I also have one that has a caption and two images following the caption, which originally did something unexpected -- it treated the two images as a singular image -- and so I updated the regular expression so that the caption will only apply to the first image and the second image will be handled as if a captionless image.
